### ISSUE TITLE
Unified hcloud experience over all hetzner products (S3 Object Storage, Dns, Storage Box, ...) 

### ISSUE THREAD

**Issue by c33s** [hetznercloud/cli#918]
### TL;DR

providing a unified, consistent experience for developers across all hetzner products would be such a win for your customers. we dream of having one tool to rule them all! 😉

### Expected behavior

> the DNS API is not Part of the Hetzner Cloud product. Therefore there are no plans to integrate the DNS API into the hcloud CLI.

https://github.com/hetznercloud/cli/issues/321#issuecomment-830798363

> All our integrations rely on the Hetzner Cloud public API, which is available with a certain level of stability. Since the features you are requesting are not in the public API, we cannot implement them.

https://github.com/hetznercloud/cli/issues/880#issuecomment-2476414492

as a developer, i don’t really mind which product an api belongs to, as long as it’s from the same provider. ideally, these details (which department or team is responsible) should be managed internally. from our perspective, we just want seamless support across **all** hetzner products.

for many of us, hcloud is the cli tool for interacting with hetzner, and expanding its capabilities to cover more products (like dns, s3, storage box, robot, etc.) would be fantastic! making your products more accessible to developers will only encourage wider adoption—and ultimately benefit both sides. for instance, integrating storage box with terraform would be amazing!

it’s your "fault" that i even ask about hcloud/terraform support, you created such a great experience with hcloud (good listing, excellent export possibilities, great terraform provider, ...) that working with a different product of hetzner can easily feels like pain.. 
hetzner/you brought me to terraform, i read a newsletter announcement you made or the cloud documentation (i cannot remember) and read about terraform. since then i started using terraform and choose only providers which have a good api with terraform support. 
you don't have to wonder developers ask for this, your tool is just too good  😛😀

providing a unified, consistent experience for developers across all hetzner products would be such a win for your customers. we dream of having one tool to rule them all! 😉

**Comment by bebehei** [comment]
This is totally understandable. But Hetzner is divided in both Hetzner (the Bare-Metal/Robot/StorageBox stuff) and Hetzner Cloud (VMs/Load Balancer/Object Storage stuff). These are even two companies. By [conway's law](https://en.wikipedia.org/wiki/Conway%27s_law), you're requiring a full reorganisation Hetzner with this ticket.

Still, I'm totally rooting for it!

**Comment by mzhaase** [comment]
Well even object storage is not in their terraform provider. I am an infrastructure consultant and when I float the idea of migrating from US providers to hetzner, friction points like these are still what makes them stay with AWS. From the outside it seems hetzner is still not really taking "that cloud stuff" seriously.

**Comment by tkschmidt** [comment]
I have to agree. I was planing on migrating parts from AWS to hetzner, but not having a terraform setup that works for both (vms and the object storage) is a big no-go.

**Comment by pat-s** [comment]
@tkschmidt While there is no unified one, TF providers for both servers and object storage exist and are documented on the Hetzner website.

**Comment by rwunderer** [comment]
@mzhaase , @tkschmidt I totally agree with @pat-s here: In any reasonably sized TF project you need to mix a multitude of providers. And terraform (or opentofu) provides a nice unified experience across your project where you can e.g. use the output of one resource as input to another regardless of the providers involved. 👍 

That said it would be nice to be able to provision bare metal servers with terraform, but I wouldn't mind if that was a separate provider. 😃

**Comment by c33s** [comment]
unstale

**Comment by dorianborovina** [comment]
Unstale and upvote. 👍🏻 
Hetzner S3 API is the only one missing right now.
